src/wp-includes/class-wp-script-modules.php

```diff
@@ -262,14 +272,54 @@ public function print_import_map() {
 	 *
 	 * @since 6.5.0
 	 *
-	 * @return array Array with an `imports` key mapping to an array of script module identifiers and their respective
-	 *               URLs, including the version query.
+	 * @global WP_Dependencies $wp_scripts
+	 *
+	 * @return array Array with an `imports` key mapping to an array of script
+	 *               module identifiers and their respective URLs, including
+	 *               the version query.
 	 */
 	private function get_import_map(): array {
+		global $wp_scripts;
+		$script_module_ids = array();
+		if ( $wp_scripts instanceof WP_Dependencies ) {
+			foreach ( $wp_scripts->registered as $dependency ) {
+				$handle = $dependency->handle;
+
+				if (
+					! $wp_scripts->query( $handle, 'done' ) &&
+					! $wp_scripts->query( $handle, 'to_do' ) &&
+					! $wp_scripts->query( $handle, 'enqueued' )
+				) {
+					continue;
+				}
```

Comment:
Is it sufficient to just check `enqueued`?

```suggestion
				if ( ! $wp_scripts->query( $handle, 'enqueued' ) ) {
					continue;
				}
```

Reply:
Apparently yes. It seems that `$wp_scripts->queue` is never emptied out except by `WP_Dependencies::dequeue()`. WordPress knows to not re-print a script because it is added to `$wp_scripts->done`.

Reply:
Oh, maybe not, because this is not accounting for the dependencies of a queued script. So I think you do need to check `to_do` and `done` as well.